doc/source/serve/model_composition.md

```diff
@@ -72,8 +79,53 @@ Composition lets you break apart your application and independently scale each p
 With composition, you can avoid application-level bottlenecks when serving models and business logic steps that use different types and amounts of resources.
 :::
 
+### ServeHandle Deep Dive
+
+Conceptually, ServeHandle is a client side load balancer, routing requests to any replicas of a given deployment. Also, it performs buffering internally so it won't overwhelm the replicas.
```

Comment:
> Conceptually, ServeHandle is a client side load balancer

"client" is a bit vague here. At first glance, it might refer to the end user (not the deployment running in the cluster). Is there a synonym we could use?

Reply:
"client-side load balancer" is a well known term